onnxruntime/python/tools/quantization/test_calibrate.py

```diff
@@ -1,24 +1,103 @@
+#!/usr/bin/env python
+# coding: utf-8
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
 import unittest
+import os
 import onnx
-
-from quantize import quantize, QuantizationMode
+import onnxruntime
+from onnx import helper, TensorProto
+from onnx.helper import make_node, make_tensor_value_info
 import calibrate
 
-class ProcessLogFileTest(unittest.TestCase):
-    # def test_baddir(self):
-    #     self.assertRaises(ValueError, calibrate.process_logfiles('/non/existent/path'))
+class TestCalibrate(unittest.TestCase):
+
+    def test_augment_graph(self):
+        # Creating graph
+        A = helper.make_tensor_value_info('A', TensorProto.FLOAT, [1, 1, 5, 5])
+        B = helper.make_tensor_value_info('B', TensorProto.FLOAT, [1, 1, 3, 3])
+        C = helper.make_tensor_value_info('C', TensorProto.FLOAT, [1, 1, 5, 5])
+        D = helper.make_tensor_value_info('D', TensorProto.FLOAT, [1, 1, 5, 5])
+        E = helper.make_tensor_value_info('E', TensorProto.FLOAT, [1, 1, 5, 1])
+        F = helper.make_tensor_value_info('F', TensorProto.FLOAT, [1, 1, 5, 1])
+        conv_node = onnx.helper.make_node('Conv', ['A', 'B'], ['C'], name='Conv', kernel_shape=[3, 3], pads=[1, 1, 1, 1])
+        clip_node = onnx.helper.make_node('Clip', ['C'], ['D'], name='Clip')
+        matmul_node = onnx.helper.make_node('MatMul', ['D', 'E'], ['F'], name='MatMul')
+        graph = helper.make_graph([conv_node, clip_node, matmul_node], 'test_graph', [A, B, E], [F])
+        model = helper.make_model(graph)
+        onnx.save(model, 'test_model.onnx')
+
+        # Augmenting graph
+        augmented_model = calibrate.augment_graph(model)
+        onnx.save(augmented_model, 'augmented_test_model.onnx')
+
+        # Checking if each added ReduceMin and ReduceMax node and its output exists
+        augmented_model_node_names = [node.name for node in augmented_model.graph.node]
+        augmented_model_outputs = [output.name for output in augmented_model.graph.output]
+        added_node_names = ['Conv_ReduceMin', 'Conv_ReduceMax', 'MatMul_ReduceMin', 'MatMul_ReduceMax']
+        added_outputs = ['C_ReduceMin', 'C_ReduceMax', 'F_ReduceMin', 'F_ReduceMax']
+        self.assertEqual(len(augmented_model_node_names), 7) # original 3 nodes with 4 added ones
+        self.assertEqual(len(augmented_model_outputs), 5) # original single graph output with 4 added ones
+        for name in added_node_names:
+            self.assertTrue(name in augmented_model_node_names)
+        for output in added_outputs:
```

Comment:
checking order is equally important.

Reply:
Testing for the expected order is encompassed within the way the added output is named in calibrate.py (whatever is in front of '_ReduceMin' or '_ReduceMax' is the corresponding augmented node's output); should the check for order be more explicit?

Reply:
Resolved this and added an additional test.